app/components/curriculum-inventory-report-header.js

```diff
@@ -17,16 +20,54 @@ const Validations = buildValidations({
 });
 
 export default Component.extend(Validations, ValidationErrorDisplay, {
+  flashMessages: service(),
+
   didReceiveAttrs(){
     this._super(...arguments);
     this.set('reportName', this.get('report.name'));
     this.set('isFinalized', this.get('report.isFinalized'));
+
+    // check if the d/l link points at the same domain as the current page is on.
+    let parser = document.createElement('a');
+    parser.href = this.get('report.absoluteFileUri');
+    let backendDomain = parser.hostname;
+    let frontendDomain = window.location.hostname;
+    this.set('downloadFromSameDomain', (backendDomain === frontendDomain));
```

Comment:
What about if just the `absoluteFileUri` changes? It isn't a very likely change given that it is generated on the server.  But if some kind of future enhancement updated that field then the download wouldn't work.  That is where I got into trouble with this on the user profile page. Changes to properties on the object don't trigger an update.